File: app/src/main/java/com/gongbo/fss/demo/adapter/AdapterActivity.java
package com.gongbo.fss.demo.adapter;

import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import com.gongbo.fss.adapter.listview.CommonAdapter;
import com.gongbo.fss.base.BaseFssActivity;
import com.gongbo.fss.bind.annotation.BindActivity;
import com.gongbo.fss.bind.annotation.BindView;
import com.gongbo.fss.demo.R;
import com.gongbo.fss.demo.base.BaseActivity;
import com.gongbo.fss.router.FssRouteApi;
import com.gongbo.fss.router.annotation.Route;

import java.util.Arrays;
import java.util.List;



@Route
@BindActivity(value = R.layout.activity_adapter, finishViewId = R.id.img_back)
public class AdapterActivity extends BaseActivity implements AdapterView.OnItemClickListener {

    @BindView(R.id.list_view)
    private ListView listView;

    private List<String> datas = Arrays.asList(
            "BaseAdapter",
            "BaseBindAdapter",
            "BaseBindingAdapter",
            "BaseBindingSingleAdapter",
            "CommonAdapter",
            "BaseAdapter",
            "BaseBindRecyclerViewAdapter",
            "BaseBindingAdapter",
            "CommonRecyclerAdapter"
    );

    protected void initView() {
        listView.setAdapter(new CommonAdapter<>(this, datas, R.layout.layout_list_item,
                (holder, s, position) -> holder.setText(R.id.tv_text, s)));
        listView.setOnItemClickListener(this);
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
        switch (position) {
            case 0:
                FssRouteApi.LIST_VIEW.navigateToBaseAdapterTestActivity(this);
                break;
            case 1:
                FssRouteApi.LIST_VIEW.navigateToBaseBindAdapterTestActivity(this);
                break;
            case 2:
                FssRouteApi.LIST_VIEW.navigateToBaseBindingAdapterTestActivity(this);
                break;
            case 3:
                FssRouteApi.LIST_VIEW.navigateToBaseBindingSingleAdapterTestActivity(this);
                break;
            case 4:
                FssRouteApi.LIST_VIEW.navigateToCommonAdapterTestActivity(this);
                break;
            case 5:
                FssRouteApi.RECYCLER_VIEW.navigateToBaseRecyclerViewAdapterTestActivity(this);
                break;
            case 6:
                FssRouteApi.RECYCLER_VIEW.navigateToBaseBindRecyclerViewAdapterTestActivity(this);
                break;
            case 7:
                FssRouteApi.RECYCLER_VIEW.navigateToBaseBindingRecyclerViewAdapterTestActivity(this);
                break;
            case 8:
                FssRouteApi.RECYCLER_VIEW.navigateToCommonRecyclerAdapterViewTestActivity(this);
                break;
            default:
                throw new IllegalStateException("Unexpected value: " + position);
        }
    }

}

File: app/src/main/java/com/gongbo/fss/demo/adapter/AdapterActivity.kt
package com.gongbo.fss.demo.adapter

import android.view.View
import android.widget.AdapterView
import android.widget.ListView

import com.gongbo.fss.adapter.listview.CommonAdapter
import com.gongbo.fss.base.BaseFssActivity
import com.gongbo.fss.bind.annotation.BindActivity
import com.gongbo.fss.bind.annotation.BindView
import com.gongbo.fss.demo.R
import com.gongbo.fss.router.annotation.Route

import java.util.Arrays

import com.gongbo.fss.router.FssRouteApi.ListViewRouteApi
import com.gongbo.fss.router.FssRouteApi.RecyclerViewRouteApi


@Route
@BindActivity(layout = R.layout.activity_adapter, finish = [R.id.img_back])
class AdapterActivity : BaseFssActivity(), AdapterView.OnItemClickListener {

    @BindView(id = R.id.list_view)
    private val listView: ListView? = null

    private val datas = Arrays.asList(
            "BaseAdapter",
            "BaseBindAdapter",
            "BaseBindingAdapter",
            "BaseBindingSingleAdapter",
            "CommonAdapter",
            "BaseAdapter",
            "BaseBindRecyclerViewAdapter",
            "BaseBindingAdapter",
            "CommonRecyclerAdapter"
    )

    override fun initView() {
        super.initView()
        listView!!.adapter = CommonAdapter(this, datas, R.layout.layout_list_item
        ) { holder, s, _ -> holder.setText(R.id.tv_text, s) }
        listView.onItemClickListener = this
    }

    override fun onItemClick(parent: AdapterView<*>, view: View, position: Int, id: Long) {
        when (position) {
            0 -> ListViewRouteApi.navigateToBaseAdapterTestActivity(this)
            1 -> ListViewRouteApi.navigateToBaseBindAdapterTestActivity(this)
            2 -> ListViewRouteApi.navigateToBaseBindingAdapterTestActivity(this)
            3 -> ListViewRouteApi.navigateToBaseBindingSingleAdapterTestActivity(this)
            4 -> ListViewRouteApi.navigateToCommonAdapterTestActivity(this)
            5 -> RecyclerViewRouteApi.navigateToBaseRecyclerViewAdapterTestActivity(this)
            6 -> RecyclerViewRouteApi.navigateToBaseBindRecyclerViewAdapterTestActivity(this)
            7 -> RecyclerViewRouteApi.navigateToBaseBindingRecyclerViewAdapterTestActivity(this)
            8 -> RecyclerViewRouteApi.navigateToCommonRecyclerAdapterViewTestActivity(this)
            else -> throw IllegalStateException("Unexpected value: $position")
        }
    }

}

